### PR TITLE
quic: increasing upstream packet limits

### PR DESCRIPTION
**Pull request by alyssawilk**
Debugging #16853 it turned out that since we added read limits for packets, TCP downstreams were vastly out-sending UDP upstreams.  Changing the upstream limit to be a factor of streams rather than flat rate.

Risk Level: low (H3 only)
Testing: new unit tests
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: source/common/quic/envoy_quic_client_session.h [hunk 1]
```diff
@@ -69,6 +70,12 @@ class EnvoyQuicClientSession : public QuicFilterManagerConnectionImpl,
   // quic::QuicSpdyClientSessionBase
   void SetDefaultEncryptionLevel(quic::EncryptionLevel level) override;
 
+  // PacketsToReadDelegate
+  size_t numPacketsExpectedPerEventLoop() override {
+    // Just as TCP does 16k per stream, do this for UDP.
+    return std::max<size_t>(1, GetNumActiveStreams()) * 16;
```
Comment (danzh2010) [inline source/common/quic/envoy_quic_client_session.h:76]: The magic number 16 is from TCP upstream connection? Can we use a const for both plances?

**Review by danzh2010 — APPROVED** [review]
Thanks for fixing this! LGTM along aside a nit.

**Review by antoniovicente — APPROVED** [review]
Thanks!

File: source/common/quic/envoy_quic_client_connection.h [hunk 2]
```diff
@@ -54,11 +61,14 @@ class EnvoyQuicClientConnection : public quic::QuicConnection,
     // TODO(mattklein123): Emit a stat for this.
   }
   size_t numPacketsExpectedPerEventLoop() const override {
+    if (delegate_.has_value()) {
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply source/common/quic/envoy_quic_client_connection.h:64]: yeah, the connection exists before setUpConnectionSocket but AFIK it's a bug to use or migrate it before it's called.
Can't hurt to make both optref to handle such a bug without crashing though, updated.

**Review by antoniovicente — APPROVED** [review]

**Pull request merged**